### PR TITLE
samples: matter: make device types conform with the spec

### PR DESCRIPTION
**Pull request by Damian-Nordic**
Update ZAP configuration of Matter samples so that it includes all required cluster and cluster elements for given device types. Also, align the root endpoint configuration with Matter Weather Station which passed Matter 1.0 SVE.

The change depends on and includes https://github.com/nrfconnect/sdk-nrf/pull/9002. Turn off displaying the first commit when reviewing.

### CODE REVIEW

File: applications/matter_weather_station/README.rst
Comment (greg-fer) [inline applications/matter_weather_station/README.rst:17]: Apply edits to this file as discussed in https://github.com/nrfconnect/sdk-nrf/pull/9002 from @kkasperczyk-no

**Review by greg-fer — DISMISSED** [review]

File: applications/matter_weather_station/README.rst
Comment (kkasperczyk-no) [reply applications/matter_weather_station/README.rst:17]: @greg-fer I believe this PR will be rebased after merging #9002 and my commit was pulled only to provide some base for testing Damian changes, so maybe modifying it is not needed.

Comment (greg-fer) [reply applications/matter_weather_station/README.rst:17]: Ah ok, mysteries of the GitGods!

**Review by greg-fer — COMMENTED** [review]
Just make sure to rebase to include changes from @kkasperczyk-no 's PR. I don't think my approval is needed, as these doc changes were approved elsewhere.

File: applications/matter_weather_station/README.rst
Comment (Damian-Nordic) [reply applications/matter_weather_station/README.rst]: Rebased. Please re-review. @greg-fer - yes, I guess I still need your approval although you already reviewed it on the other branch.

**Review by kkasperczyk-no — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**
